### PR TITLE
fix(releases): For semver releases, get latest release as the resolving one

### PR DESCRIPTION
**Pull request by armenzg**
Select the latest semver release for the project rather than the latest seen for the group.

This is a follow-up to https://github.com/getsentry/sentry/pull/80655.
This fixes the customer issue https://github.com/getsentry/sentry/issues/69937 (and possibly these: #74448 and #80194).

### CODE REVIEW

File: src/sentry/api/helpers/group_index/update.py [hunk 1]
```diff
@@ -419,9 +416,19 @@ def update_groups(
...
-                            group=group, follows_semver=follows_semver
-                        )
+                        if (
+                            features.has(
+                                "organizations:releases-resolve-next-release-semver-fix",
+                                project.organization,
+                            )
+                            and follows_semver
+                        ):
+                            current_release_version = get_latest_release(projects[0]).version
```
Comment (armenzg) [inline src/sentry/api/helpers/group_index/update.py:426]: This is the actual fix.

File: tests/sentry/issues/endpoints/test_group_details.py [hunk 2]
```diff
@@ -407,17 +448,45 @@ def test_resolved_in_next_release_semver(self):
...
+        # thus, it is considered resolved in the release that has the highest semver
+        assert group_resolution.type == GroupResolution.Type.in_release
+        assert group_resolution.status == GroupResolution.Status.resolved
         assert group_resolution.release.version == greatest_version.version
+        assert response.data["statusDetails"]["inRelease"] == greatest_version.version
+
+        new_release = Release.get_or_create(version="com.foo.bar@3.0+0", project=project)
+        # A lesser release than 2.x but created more recently than 3.x
+        old_version = Release.get_or_create(version="com.foo.bar@1.1+0", project=project)
+        for release in releases + [old_version]:
```
Comment (armenzg) [inline tests/sentry/issues/endpoints/test_group_details.py:461]: Without the fix, only `com.foo.bar@1.0+0` would _not_ unresolve.

**Review by snigdhas — APPROVED** [review]
Looks great!

**Review by JoshFerge — APPROVED** [review]

File: src/sentry/api/helpers/group_index/update.py [hunk 3]
```diff
@@ -419,9 +416,19 @@ def update_groups(
                             release_version=release.version,
                         )
 
-                        current_release_version = get_current_release_version_of_group(
```
Comment (wedamija) [inline src/sentry/api/helpers/group_index/update.py:422]: I think we should look to remove this as a follow-up https://github.com/getsentry/sentry/blob/c0797226ec0dc304d766796713121a880df38811/src/sentry/integrations/tasks/sync_status_inbound.py#L112-L114 and I think that actually that code path should have the same fix you're adding in here. So might be worth also adding the fix there as part of this

Comment (armenzg) [reply src/sentry/api/helpers/group_index/update.py:422]: Definitely. I was trying to mostly working and not being awful code to review.

Comment (armenzg) [reply src/sentry/api/helpers/group_index/update.py:422]: Sounds good!

Comment (armenzg) [reply src/sentry/api/helpers/group_index/update.py:422]: I followed-up here: https://github.com/getsentry/sentry/pull/80821

**Comment by armenzg** [comment]
I will enable this for GA next week.

**Pull request merged**